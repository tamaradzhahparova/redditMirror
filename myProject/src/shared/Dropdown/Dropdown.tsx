import React, { FC, useEffect, useState} from 'react';
import styles from './Dropdown.module.css';

interface DropdownProps {
  button: React.ReactNode
  children: React.ReactNode
  isOpen?: boolean
  onOpen?: () => void
  onClose?: () => void
}

const noop = () => {
}

const Dropdown: FC<DropdownProps> = ({button, children, onOpen = noop, isOpen, onClose = noop}) => {
  
  const [isDropdownOpen, setIsDropdownOpen] = useState(isOpen)
  
  useEffect(() => setIsDropdownOpen(isOpen), [isOpen])
  useEffect(() => isDropdownOpen ? onOpen() : onClose(), [isDropdownOpen])
  
  const handleOpen = () => {
    if (isOpen === undefined) {
      setIsDropdownOpen(!isDropdownOpen)
    }
  }
  
  
  return (
    <div className={styles.Dropdown}>
      <div onClick={handleOpen}>
        {button}
      </div>
      {isDropdownOpen && (
        <div className={styles.listContainer}>
          <div
            onClick={() => setIsDropdownOpen(false)}
            className={styles.list}>
            {children}
          </div>
        </div>
      )}
    </div>)
};

export default Dropdown;
