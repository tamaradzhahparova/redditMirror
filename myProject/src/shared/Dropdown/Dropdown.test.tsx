import {shallow} from "enzyme";
import React from "react";
import Dropdown from "./Dropdown";

describe('dropdown', () => {
  test('should render', () => {
    const wrapper = shallow(<Dropdown button={<button/>} children={<div/>}/>)
    expect(wrapper).toBeDefined()
    expect(wrapper.find('div.Dropdown').isEmptyRender()).toBeFalsy()
  })

  test('should render (snapshoot)', () => {
    const wrapper = shallow(<Dropdown button={<button/>} children={<div/>}/>)
    expect(wrapper).toMatchSnapshot()
  })
})