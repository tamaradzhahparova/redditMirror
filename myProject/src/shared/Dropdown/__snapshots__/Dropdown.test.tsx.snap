// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown should render (snapshoot) 1`] = `
<div
  className="Dropdown"
>
  <div
    onClick={[Function]}
  >
    <button />
  </div>
</div>
`;
